packages/stores/src/account/osmosis/index.ts

```diff
@@ -2040,6 +2039,130 @@ export class OsmosisAccountImpl {
     );
   }
 
+  /**
+   * Method to undelegate from validator set.
+   * @param delegator The user who is trying to undelegate.
+   * @param coin The coin object with denom and amount to undelegate.
+   * @param memo Transaction memo.
+   * @param stdFee Fee options.
+   * @param signOptions Signing options.
+   * @param onFulfill Callback to handle tx fulfillment given raw response.
+   */
+  async sendUndelegateFromValidatorSetMsg(
+    delegator: string,
+    coin: { currency: Currency; amount: string },
+    memo: string = "",
+    stdFee: Partial<StdFee> = {},
+    signOptions?: KeplrSignOptions,
+    onFulfill?: (tx: any) => void
+  ) {
+    await this.base.signAndBroadcast(
+      this.chainId,
+      "undelegateFromValidatorSet",
+      async () => {
+        const coinAmount = new Dec(coin.amount)
+          .mul(
+            DecUtils.getTenExponentNInPrecisionRange(coin.currency.coinDecimals)
+          )
+          .truncate();
+        const cosmosCoin = new Coin(coin.currency.coinMinimalDenom, coinAmount);
+
+        const msg = this.msgOpts.undelegateFromValidatorSet.messageComposer({
+          delegator: delegator,
+          coin: {
+            denom: cosmosCoin.denom,
+            amount: cosmosCoin.amount.toString(),
+          },
+        });
+
+        return [msg];
+      },
+      memo,
+      {
+        amount: [],
+        gas: this.msgOpts.undelegateFromValidatorSet.gas.toString(),
+        ...stdFee,
+      },
+      signOptions,
+      (tx) => {
+        if (tx.code == null || tx.code === 0) {
+          // Refresh the balances
+          const queries = this.queriesStore.get(this.chainId);
+          queries.queryBalances
+            .getQueryBech32Address(this.address)
+            .balances.forEach((bal) => {
+              if (
+                bal.currency.coinMinimalDenom === coin.currency.coinMinimalDenom
+              ) {
+                bal.waitFreshResponse();
+              }
+            });
+
+          queries.osmosis?.queryAccountsPositions
+            .get(this.address)
+            .waitFreshResponse();
+        }
+        onFulfill?.(tx);
+      }
+    );
+  }
+
+  /**
+   * Method to delegate to validator set.
+   * @param delegator The user who is trying to delegate.
+   * @param memo Transaction memo.
+   * @param stdFee Fee options.
+   * @param signOptions Signing options.
+   * @param onFulfill Callback to handle tx fulfillment given raw response.
+   */
+  async sendDelegateToValidatorSetMsg(
+    delegator: string,
+    coin: Coin, // Use the Coin object from "@keplr-wallet/unit"
```

Comment:
perhaps define type manually above and use here instead of using keplr wallet type